### DISCUSSION TITLE
How can I save 64bit PNG as 32bit PNG?

### DISCUSSION THREAD

**Discussion by FurkanGozukara** [dlemstra/Magick.NET#982] [Help]
This is the command of imagemagick but how can I do same with Magick.NET?

`convert input.png png32:output.png`

**Comment by dlemstra** [comment]
You can do this with using `MagickFormat.Png32` as the format of your output file.

**Reply by FurkanGozukara** [reply]
thank you very much

i made like this and it also worked would it be any different?
```

FileInfo imgInfo = new FileInfo(myFile.FullName);

                using (MagickImage img = new MagickImage(imgInfo))
                {
                    if(img.BitDepth()>8)
                    {
                        img.BitDepth(8);
                        img.Write(imgInfo);
                    }                
                }
```

**Reply by dlemstra** [reply]
This probably also fixes it but without `MagickFormat.Png32` the output image could also be a 1 bit image.

**Reply by FurkanGozukara** [reply]
@dlemstra why it can be also a 1 bit image? what is the difference? my method wouldn't reduce image quality am i right?

**Reply by dlemstra** [reply]
The PNG coder could decide that the image can be saved in a lower bit depth. This behavior is different per coder.

**Reply by FurkanGozukara** [reply]
@dlemstra ah that is even better as long as no quality is lost
